Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.Firestore.DistributedCache/FirestoreCache.cs

```diff
@@ -0,0 +1,323 @@
+﻿// Copyright (c) 2019 Google LLC.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Apis.Auth.OAuth2;
+using Google.Cloud.Firestore;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Logging;
+
+namespace Google.Cloud.AspNetCore.Firestore.DistributedCache
+{
+    /// <summary>
+    /// A distributed cache that stores entries in Firestore.
+    /// </summary>
+    public class FirestoreCache : IDistributedCache
+    {
+        private readonly FirestoreDb _firestore;

```

Comment:
I would prefer to call the `FirestoreDb` either `_firestoreDb` or `_db` (with constructor parameter names to match, etc.). I acknowledge it's a bit different from the convention for other languages (where the class is just called "Firestore") but it feels more appropriate to me. It's not a massively strong opinion though.

Reply:
Done.